### DISCUSSION TITLE
Linked or parent project

### DISCUSSION THREAD

**Discussion by manuelcanepa** [wardenenv/warden#756] [Help & Support]
Hi everyone, there is any way to create an environment with a dependency? This is what i think:

```
- project1
  - .env
- project2
  - .env
```

```
# project1/.env
WARDEN_ENV_NAME=project1
WARDEN_ENV_TYPE=local
WARDEN_WEB_ROOT=/

TRAEFIK_DOMAIN=project1.test
TRAEFIK_SUBDOMAIN=app
```

```
# project2/.env
WARDEN_ENV_NAME=project2
WARDEN_ENV_TYPE=local
WARDEN_WEB_ROOT=/
WARDEN_DEPENDS=../project1  # Here is the magic

TRAEFIK_DOMAIN=project2.test
TRAEFIK_SUBDOMAIN=app
```

In this example, if you run `warden env up` on project1 directory, will only start the project1, however, if you run `warden env up` on project2 will start first project1 and then project2.

Is it possible?

**Comment by bap14** [comment]
This isn't currently possible; however, I am interested in your use-case for this.

The immediate difficulty I see in this is how to handle a dependent project starting but then you manually stopping it. Does the "child" project that depends on it also need to stop? What if you are stopping it to start it again? Do we need to know that the child project was running but stopped because the parent project was stopped and now that it's restarted we'll restart the child?

One option could be to use a custom `healthcheck` for one or more of the primary containers (e.g. php-fpm) which will check for the dependent service and "fail" if it's not available. This would at least prevent you from starting it without the dependent project; however, there won't be any error messages stating to start a new project.
